parser-library/buffer.cpp

```diff
@@ -52,6 +52,22 @@ struct buffer_detail {
 #endif
 };
 
+#ifndef WIN32
```

Comment:
I think here the static may not be necessary as well if we put the functions into an anonymous namespace:
http://stackoverflow.com/questions/154469/unnamed-anonymous-namespaces-vs-static-functions